### PR TITLE
call core api delete from django delete

### PR DESCRIPTION
**Pull request by rachaelcodes**
## Context

Currently the frontend deletion of a file only deletes the file from the Django and S3 storage, but not the Elastic storage.

## Changes proposed in this pull request
Deleting a file from the user interface propagates a call to core-api to delete the file and its chunks. If that API request fails, then the file will not delete from the Django database.

If the call to the API fails, the user will see this view: 
Screenshot 2024-05-20 at 12 31 30

## Guidance to review

Try deleting files from django.

## Relevant links
https://technologyprogramme.atlassian.net/browse/REDBOX-256

## Things to check

- ~[ ] I have added any new ENV vars in all deployed environments~
- [x] I have tested any code added or changed
- [ ] I have run integration tests

### CODE REVIEW

File: django_app/tests/test_views.py [hunk 1]
```diff
@@ -195,7 +238,6 @@ def test_post_message_to_new_session(alice: User, client: Client, requests_mock:
     assert response.status_code == HTTPStatus.FOUND
     assert "Location" in response.headers
     session_id = URL(response.url).parts[-2]
-    assert ChatMessage.objects.get(chat_history__id=session_id, role=ChatRoleEnum.user).text == "Are you there?"
```
Comment (rachaelcodes) [inline django_app/tests/test_views.py:198]: hm not sure why this got deleted - will fix on Monday

File: django_app/redbox_app/redbox_core/views.py [hunk 2]
```diff
@@ -149,14 +149,27 @@ def ingest_file(uploaded_file: UploadedFile, user: User) -> list[str]:
 @login_required
 def remove_doc_view(request, doc_id: uuid):
     file = File.objects.get(pk=doc_id)
+    errors: list[str] = []
+
     if request.method == "POST":
-        logger.info("Removing document: %s", request.POST["doc_id"])
-        file.delete()
+        api = CoreApiClient(host=settings.CORE_API_HOST, port=settings.CORE_API_PORT)
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/views.py:155]: this could be declared just once, maybe as a constant?

Comment (brunns) [reply django_app/redbox_app/redbox_core/views.py]: Ultimately I'd like to inject it.

**Review by 252afh — APPROVED** [review]
Looks good to me, could we change the file deletion setting to seconds instead of days for future proofing please?

File: django_app/redbox_app/redbox_core/models.py [hunk 3]
```diff
@@ -64,6 +69,7 @@ class File(UUIDPrimaryKeyBase, TimeStampedModel):
     user = models.ForeignKey(User, on_delete=models.CASCADE)
     original_file_name = models.TextField(max_length=2048, blank=True, null=True)
     core_file_uuid = models.UUIDField(null=True)
+    expiry_date = models.DateField(default=get_default_expiry_date)
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/models.py:72]: either:
* this needs to be nullable or we will get an error on migration, or
* we could compute this field on the fly from the `created_at: datetime`
?

File: django_app/redbox_app/redbox_core/views.py [hunk 4]
```diff
@@ -24,6 +24,7 @@
 from yarl import URL
 
 logger = logging.getLogger(__name__)
+core_api = CoreApiClient(host=settings.CORE_API_HOST, port=settings.CORE_API_PORT)
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/views.py:27]: 🙏

**Review by gecBurton — CHANGES_REQUESTED** [review]
one RFC, otherwise LGTM

**Review by gecBurton — APPROVED** [review]

**Pull request merged**